src/cmake/AxomOptions.cmake

```diff
@@ -27,3 +27,12 @@ option(AXOM_ENABLE_TOOLS "Enables Axom Tools" ON)
 
 cmake_dependent_option(AXOM_ENABLE_MPI3 "Enables use of MPI-3 features" OFF "ENABLE_MPI" OFF)
 mark_as_advanced(AXOM_ENABLE_MPI3)
+
+#--------------------------------------------------------------------------
+# Option to control whether AXOM_DEFINE compiler define is enabled
+#
+# Possible values are: "ON", "OFF" and "DEFAULT"
+# By default, AXOM_DEBUG is defined in Debug and RelWithDebInfo configurations
+#--------------------------------------------------------------------------
+set(AXOM_DEBUG_DEFINE "DEFAULT" CACHE STRING "Controls whether AXOM_DEBUG compiler define is enabled")
+set_property(CACHE AXOM_DEBUG_DEFINE PROPERTY STRINGS "DEFAULT" "ON" "OFF")
```

Comment:
This adds the new config option: `AXOM_DEBUG_DEFINE`, and defines the three valid options.